lib/node_modules/@stdlib/blas/ext/base/dnannsumpw/src/main.c

```diff
@@ -148,21 +151,19 @@ static double sumpw( const int64_t N, const double *X, const int64_t stride, int
 				sum += v;
 				*n += 1;
 			}
-			ix += stride;
+			ix += strideX;
 		}
 		return sum;
 	}
 	// Recurse by dividing by two, but avoiding non-multiples of unroll factor...
 	m = N / 2;
 	m -= m % 8;
-	if ( stride < 0 ) {
-		xp1 = (double *)X + ( (m-N)*stride );
-		xp2 = (double *)X;
-	} else {
-		xp1 = (double *)X;
-		xp2 = (double *)X + ( m*stride );
-	}
-	return sumpw( m, xp1, stride, n ) + sumpw( N-m, xp2, stride, n );
```

Comment:
@headlessNode Thanks for catching. This looks to have been a bug in the original implementation.

Reply:
Oh, I see. We needed to refactor because of L55, which explicitly sets the value to `0`, which previously was done in the calling function.

Reply:
Is this change necessary? Couldn't we explicitly set `*n = 0` at L187 and L210 and avoid the allocation of 4 additional variables on the stack?

Reply:
Yes the pointer can accumulate in through the recursive calls. Thanks for pointing this out.